projects/demo/src/components/control/index.ts

```diff
@@ -0,0 +1,25 @@
+import {NgIf} from '@angular/common';
+import {ChangeDetectionStrategy, Component, Input} from '@angular/core';
+import type {AbstractControl} from '@angular/forms';
+import {DemoRoute} from '@demo/routes';
+import {TuiDocAPIItem} from '@taiga-ui/addon-doc';
+import type {TuiLooseUnion} from '@taiga-ui/cdk';
+import {TuiTitle} from '@taiga-ui/core';
+
+@Component({
+    standalone: true,
+    selector: 'tbody[tuiDocControl]',
+    imports: [NgIf, TuiDocAPIItem, TuiTitle],
+    templateUrl: './index.html',
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+export class TuiDocControl {
+    protected readonly routes = DemoRoute;
+
+    @Input()
+    public hiddenOptions: Array<TuiLooseUnion<keyof AbstractControl>> = [];
```

Comment:
It's not properties of `AbstractControl`, it's just these 3 hardcoded values. For `Textfield` they come from here:
https://github.com/taiga-family/taiga-ui/blob/main/projects/core/components/textfield/textfield.directive.ts#L35-L39
And for controls like `InputCard` or `InputPhoneInternational` it comes from our abstract control class:
https://github.com/taiga-family/taiga-ui/blob/main/projects/cdk/classes/control.ts#L84-L89

Reply:
Yes, i seen. I think it would be cumbersome to derive these types from abstract class. 
Let's use union `'readOnly' | 'disabled' | 'invalid'` to simplify?

Reply:
Yes, that's what I think as well.